addons/sale/tests/test_sale_order_discount.py

```diff
@@ -107,3 +107,19 @@ def test_sol_discount_removal(self):
     def test_percent_discount_above_100(self):
         with self.assertRaises(ValidationError):
             self.wizard.write({'discount_percentage': 1.1, 'discount_type': 'sol_discount'})
+
+    def test_line_and_global_discount(self):
+        solines = self.sale_order.order_line
+        amount_before_discount = self.sale_order.amount_total
```

Comment:
```suggestion
        amount_before_discount = self.sale_order.amount_untaxed
```

test might fail in some cases if a localization and taxes were installed, it's safer to check always the same field ;)